logic/strategy/src/one_hop_connection.rs

```diff
@@ -0,0 +1,358 @@
+                        if outgoing_channels
+                            .clone()
+                            .any(|outgoing_c| outgoing_c.destination == relay_node)
+                        {
+                            let peers_with_quality_cloned = peers_with_quality.clone();
+                            let relay_node_quality = {
+                                let quality = peers_with_quality_cloned.get(&relay_node);
+                                quality.unwrap_or(&0_f64)
+                            };
+                            if *relay_node_quality >= self.cfg.network_quality_threshold {
+                                Some((relay_node, channel.balance))
+                            } else {
+                                None
+                            }
+                        } else {
+                            None
+                        }
+                    })
+                    .collect::<Vec<(Address, Balance)>>();
+                if relay_nodes.is_empty() {
+                    should_find_new_relay_node = true;
+                } else {
+                    should_find_new_relay_node = false;
+
+                    // Among those relay nodes with good network quality, top up channels to relay nodes with low balance.
+                    for (relay_node, outgoing_channel_to_relay_node_balance) in relay_nodes {
+                        // check if the channel needs to be topped up
+                        if outgoing_channel_to_relay_node_balance.lt(&self.cfg.min_stake_threshold) {
+                            if !tick_decision.will_address_be_opened(&relay_node) {
+                                tick_decision.add_to_open(relay_node, self.cfg.funding_amount);
+                                remaining_balance = remaining_balance.sub(&self.cfg.funding_amount);
+                                debug!(%relay_node, "relay node will be topped up");
+                            } else {
+                                debug!(%relay_node, "relay node is already promoted for topping up");
+                            }
+                        } else {
+                            // Keep existing relay nodes alive
+                            debug!(%relay_node, "relay node is alive and has enough balance");
+                        }
+                    }
+                }
+            } else {
+                should_find_new_relay_node = true;
+            }
+
+            // open a new channel to a random high quality peer, so that the destination peer has more chance of reaching this relay node
+            // and thus open a channel. Note that this assumption can be abused by malicious nodes (destination peer) to drain the node's balance.
+            // if the destination peer does not run this strategy and thus never opens a channel.
+            // This is a trade-off between security and availability. Please only use this strategy with trusted peers.
```

Comment:
The entire strategy holds this naive assumption, so the entire comment should way up next to the function definition as an official documentation.

Reply:
By the same token, it should be rather put on the README's first line in the monorepo, because not even other components, including transport, are assuming adversarial peers.